packages/customer-effort-score/src/feedback/index.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import PropTypes from 'prop-types';
+import {
+	Button,
+	Modal,
+	RadioControl,
+	__experimentalText as Text,
+	TextareaControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Provides a modal requesting customer feedback.
+ *
+ * A label is displayed in the modal asking the customer to score the
+ * difficulty completing a task. A group of radio buttons, styled with
+ * emoji facial expressions, are used to provide a score between 1 and 5.
+ *
+ * A low score triggers a comments field to appear.
+ *
+ * Upon completion, the score and comments is sent to a callback function.
+ *
+ * @param {Object} props                       Component props.
+ * @param {Function} props.recordScoreCallback Function to call when the results are sent.
+ * @param {string} props.label                 Question to ask the customer.
+ */
+function Feedback( { recordScoreCallback, label } ) {
+	const options = [
+		{
+			label: __( 'Very difficult', 'woocommerce-admin' ),
+			value: '1',
+		},
+		{
+			label: __( 'Somewhat difficult', 'woocommerce-admin' ),
+			value: '2',
+		},
+		{
+			label: __( 'Neutral', 'woocommerce-admin' ),
+			value: '3',
+		},
+		{
+			label: __( 'Somewhat easy', 'woocommerce-admin' ),
+			value: '4',
+		},
+		{
+			label: __( 'Very easy', 'woocommerce-admin' ),
+			value: '5',
+		},
+	];
+
+	const [ score, setScore ] = useState();
+	const [ comments, setComments ] = useState();
+	const [ errorMessage, setErrorMessage ] = useState();
+	const [ isOpen, setOpen ] = useState( true );
+
+	const closeModal = () => setOpen( false );
+
+	const sendScore = () => {
+		if ( ! [ '1', '2', '3', '4', '5' ].includes( score ) ) {
+			setErrorMessage(
+				__( 'Error: Feedback not provided.', 'woocommerce-admin' )
+			);
+			return;
+		}
+		setOpen( false );
+		recordScoreCallback( score, comments );
+	};
+
+	if ( ! isOpen ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			className="woocommerce-customer-effort-score"
+			title={ __( 'Please share your feedback', 'woocommerce-admin' ) }
+			onRequestClose={ closeModal }
+			shouldCloseOnClickOutside={ false }
```

Comment:
I'm a little wary of this in the edge case where the actions to close the modal are hidden in some way and the user is stuck with the screen disabled. Do you think there's another way to solve this?

Reply:
Good points raised here 👍 

Perhaps we could try to fix the resize bug upstream in the Gutenberg component instead.

I can reproduce it in the [component Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-modal--default). It seems to close when the window size hits a responsive breakpoint.

Reply:
I agree that we should look into reporting/fixing the resize bug in upstream Gutenberg -- that clearly seems broken.

And, I also feel that having this modal dismissible by clicking outside of it is not ideal. I think the change to prevent this is a good one to make. 

Having a modal like this be so easily dismissible accidentally, where we are asking the user to provide input and there is no way for the user to easily bring back up the modal, does not feel user-friendly. Users mis-click/tap.

I don't think it is too likely that both the Cancel button at the bottom of the modal, and the X at the top, are somehow inaccessible. If they are, then there are some other issues we likely should be addressing. And, even if that were to happen, I think that 99% of users would just refresh the browser page, which would remove the dialog (and, nothing would be lost, since they only get this dialog after successfully saving something).

Reply:
@mattsherman  I agree this is a good approach to take.

I do have some doubts about it since I think some users will try to click outside with an aim to close the modal quickly.

However, I feel like guarding against mis-clicks is more important taking into account there isn't a way for the user to recover and re-open the modal.

Reply:
Yeah, I think this solution works to prevent closing by accident. And good call to report the resize bug in Gutenberg so hopefully we don't run into the same issue with future modals.